From: Y Giridhar Appaji Nag <giridhar@appaji.net>
Date: Fri, 14 Dec 2007 12:10:29 +0530
Subject: Remove bin/* and the generated lcs files in the clean target.

Description: Remove bin/* and the generated lcs files in the clean target.
 echo the command used during compilation
Author: Y Giridhar Appaji Nag <giridhar@appaji.net>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
---
 Makefile.am            |  2 ++
 lib/Makefile.am        |  2 +-
 src/Makefile.am        |  8 ++++----
 test/Makefile.am       |  9 +++++----
 test/tests2.2/Makefile |  3 +++
 test/tests2.4/Makefile | 10 ++--------
 test/tests2.5/Makefile |  3 +++
 7 files changed, 20 insertions(+), 17 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 04deb17..22f78a7 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -28,6 +28,8 @@
 
 AUTOMAKE_OPTIONS = 1.5 foreign
 
+CONFIG_CLEAN_FILES = bin/Makefile
+
 binaryfixscript = ./fixBinaryDist.sh
 SUBDIRS =   src lib imports test doc
 
diff --git a/lib/Makefile.am b/lib/Makefile.am
index 39e216e..90d7fd7 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -17,7 +17,7 @@ splintlib_DATA = \
   unix.h unix.lcd unixstrict.lcd CTrait.syms CTraitGen.lcl bool.h \
   file.mts file.xh filerw.mts filerw.xh \
   lclinit.lci linux.h lslinit.lsi tainted.mts tainted.xh \
-  stdio.h stdlib.h $(UnixHeaders)
+  $(UnixHeaders)
 
 ## Include them in the distribution
 EXTRA_DIST = $(splintlib_DATA)
diff --git a/src/Makefile.am b/src/Makefile.am
index 86f467f..50bce91 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -423,7 +423,7 @@ splinttest:
 
 binDir = bin
 
-all: splint$(EXEEXT) ../$(top_builddir)/$(binDir)/splint$(EXEEXT)
+all: splint$(EXEEXT)
 
 ../$(top_builddir)/$(binDir)/splint$(EXEEXT):  splint$(EXEEXT)
 	-$(RM) -f $(top_builddir)/$(binDir)/splint$(EXEEXT)
@@ -433,11 +433,11 @@ up:
 	-rm cgrammar.c llgrammar.c mtgrammar.c signature.c cscanner.c
 	$(MAKE) clean
 
-CLEANFILES = splint$(EXEEXT)  ../$(top_builddir)/$(binDir)/splint$(EXEEXT)
+CLEANFILES = splint$(EXEEXT)  ../$(top_builddir)/$(binDir)/splint$(EXEEXT) Headers/flag_codes.gen
 
 .c.o:
-	@echo "Compiling "$<"..."; \
-	source='$<' object='$@' libtool=no \
+	@echo "$(CCDEPMODE) $(depcomp) $(COMPILE) -c `test -f $< || echo '$(srcdir)/'`$<"
+	@source='$<' object='$@' libtool=no \
 	depfile='$(DEPDIR)/$*.Po' tmpdepfile='$(DEPDIR)/$*.TPo' \
 	$(CCDEPMODE) $(depcomp) \
 	$(COMPILE) -c `test -f $< || echo '$(srcdir)/'`$<
diff --git a/test/Makefile.am b/test/Makefile.am
index d2a751a..b9716fe 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -100,9 +100,12 @@ help:
 
 .PHONY: clean-local
 clean-local:
-	-rm -f *~ #*# *.o *.lcs a.out
+	-rm -f *~ \#*\# *.o *.lcs a.out
 	-rm -f *.out
 	-rm  *.lcd *.lh
+	-cd tests2.2; $(MAKE) clean
+	-cd tests2.4; $(MAKE) clean
+	-cd tests2.5; $(MAKE) clean
 	-cd db1; $(MAKE) clean
 	-cd db2; $(MAKE) clean
 	-cd db3; $(MAKE) clean
@@ -122,7 +125,7 @@ $(SPLINT):
 ## The tests should really be re-done as shell-scripts or something... maybe
 ##  autotest could be used once it's finished.
 
-CLEANOUTPUT = $(GREP) -v "Splint 3." | $(GREP) -v "$(SPLINT)" | $(GREP) -v "^make.*\[[1-9]*\]:" | $(GREP) -v "^gmake.*\[[1-9]*\]:" |   $(GREP) -v "^gmake -e" |  $(GREP) -v "^make -e" |$(GREP) -v "config.status: creating test/Makefile" | $(GREP) -v "cd .. && " | $(GREP) -v "CONFIG_HEADERS=" | $(GREP) -v "CONFIG_FILES="
+CLEANOUTPUT = $(GREP) -v "Splint 3." | $(GREP) -v "$(SPLINT)" | $(GREP) -v "^make.*\[[1-9]*\]:" | $(GREP) -v "^gmake.*\[[1-9]*\]:" |   $(GREP) -v "^gmake -e" |  $(GREP) -v "^make -e" |$(GREP) -v "config.status: creating test/Makefile" | $(GREP) -v "cd .. && " | $(GREP) -v "CONFIG_HEADERS=" | $(GREP) -v "CONFIG_FILES=" | $(GREP) -v "^/usr/bin/make.*\[[1-9]*\]:" | $(GREP) -v "^/usr/bin/gmake.*\[[1-9]*\]:" |   $(GREP) -v "^/usr/bin/gmake -e" |  $(GREP) -v "^/usr/bin/make -e"
 
 #drl 12/07/2002 These rules for .c and .expect files don't make sense
 # and they are confusing make dist so I'm taking them out.
@@ -1596,5 +1599,3 @@ EXTRA_DIST =  ./abst_t.lcl \
                   divzero.c parentype.c \
 stringliteral.expect  stringliteral.c \
   numabstract.c sizesigns.c typeof.c   nullassign.expect 
-
-#		sizesigns.c sizesigns.expect \
diff --git a/test/tests2.2/Makefile b/test/tests2.2/Makefile
index 5573a81..5dd3056 100644
--- a/test/tests2.2/Makefile
+++ b/test/tests2.2/Makefile
@@ -48,3 +48,6 @@ rex:
 
 struct:
 	-$(SPLINT) struct.c -expect 1
+
+clean:
+	rm -f bool.lcs
diff --git a/test/tests2.4/Makefile b/test/tests2.4/Makefile
index 109c80e..f45371f 100644
--- a/test/tests2.4/Makefile
+++ b/test/tests2.4/Makefile
@@ -78,11 +78,5 @@ alignof:
 source:
 	$(SPLINT) -D DBL_MANT_DIG=25 source.c -expect 1
 
-
-
-
-
-
-
-
-
+clean:
+	rm -f subdir/main.lcs
diff --git a/test/tests2.5/Makefile b/test/tests2.5/Makefile
index cc2de35..768e010 100644
--- a/test/tests2.5/Makefile
+++ b/test/tests2.5/Makefile
@@ -44,3 +44,6 @@ immutable:
 
 impabsmodule:
 	-${SPLINT} +impabstract impabsmodule.c -expect 2
+
+clean:
+	rm -f newlint.lcd
